### PR TITLE
kconfig: add OTA symbol that FW_UPDATE depends on

### PR DESCRIPTION
**Pull request by szczys**
Add CONFIG_GOLIOTH_OTA so that the OTA functions may be used without the firmware update functions.

Firmware updates still depend on OTA functions. Platforms that use Kconfig will automatically select OTA. However, for platforms that don't use Kconfig, ota.c will build when either CONFIG_GOLIOTH_OTA or CONFIG_GOLIOTH_FW_UPDATE are selected.

This is not a panacea but there is some savings to be had if an application wants to work with OTA componentes without fw_update:
- 60 bytes of flash saved
- 640 bytes of RAM saved

Resolves https://github.com/golioth/firmware-issue-tracker/issues/709

### CODE REVIEW

File: src/Kconfig [hunk 1]
```diff
@@ -4,12 +4,13 @@
...
+    bool "Golioth Over-the-Air Update Service"
     help
-        Enable the Golioth Firmware Update service
+        Enable the Golioth OTA service to manage the transfer of OTA components between server and
+        device. See also: GOLIOTH_FW_UPDATE
 
-if GOLIOTH_FW_UPDATE
+if GOLIOTH_OTA
 
 config GOLIOTH_OTA_THREAD_STACK_SIZE
```
Comment (sam-golioth) [inline src/Kconfig:15]: This is actually used by the fw_update module, not OTA. We should have it guarded by `GOLIOTH_FW_UPDATE` instead and also rename it.

File: src/Kconfig [hunk 2]
```diff
@@ -48,7 +49,14 @@ config GOLIOTH_OTA_OBSERVATION_RETRY_MAX_DELAY_S
         backoff doubling between retry attempts. The delay value will increase until this limit is
```
Comment (sam-golioth) [inline src/Kconfig:49]: Can't comment on the exact line, but `GOLIOTH_OTA_OBSERVATION_RETRY_MAX_DELAY_S` is also used by the firmware update module, not OTA.

**Comment by szczys** [comment]
Apologies, I didn't look to see where those encapsulated symbols were used but I definitely should have. This is sorted out now.

File: src/Kconfig [hunk 3]
```diff
@@ -37,7 +32,16 @@ config GOLIOTH_OTA_MAX_NUM_COMPONENTS
...
+endif # GOLIOTH_OTA
+
+config GOLIOTH_FW_UPDATE
+    bool "Golioth Firmware Update service"
+    select GOLIOTH_OTA
+    help
+        Enable the Golioth Firmware Update service to automatically manage firmware updates
+        utilizing functionality of the Golioth OTA service.
+
+config GOLIOTH_FW_UPDATE_OBSERVATION_RETRY_MAX_DELAY_S
```
Comment (sam-golioth) [inline src/Kconfig:44]: Should this also be guarded by `if GOLIOTH_FW_UPDATE`?

**Review by sam-golioth — APPROVED** [review]

File: src/Kconfig (hunk 3 above)
Comment (szczys) [reply src/Kconfig:46]: Yes, thanks for catching!

**Pull request merged**